### PR TITLE
[OptionsResolver] Add $triggerDeprecation arg to offsetGet() method in Options interface

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Ref: https://github.com/symfony/symfony/pull/28878

### CODE REVIEW

**Comment by yceruto** [comment]
We are missing a deprecation notice in 4.4 :/ this is a special case.

I think `DebugClassLoader` should do the job and trigger the auto-deprecation notice when the arguments defined in the `@method` annotation don't match the arguments of the real method, WDYT?

**Comment by yceruto** [comment]
We lack a deprecation notice in 4.4 :/ this is a special case.

I think `DebugClassLoader` can do the job and trigger the auto-deprecation notice when the arguments defined in the `@method` annotation don't match the arguments of the real method, WDYT?

**Comment by yceruto** [comment]
Or at least trigger only if the number of arguments in the real method is less than the defined ones in `@method` annotation

**Comment by nicolas-grekas** [comment]
I agree `DebugClassLoader` should be able to trigger a deprecation notice here. Would you be able to have a look?

**Comment by nicolas-grekas** [comment]
(don't forget the UPGRADE files :) )

**Comment by yceruto** [comment]
> I agree DebugClassLoader should be able to trigger a deprecation notice here. Would you be able to have a look?

Yes, I think so :)
Next, I need a way to get rid of that deprecation, thoughts?

**Comment by nicolas-grekas** [comment]
Adding `@param bool $triggerDeprecation` works already for getting rid of the deprecation I think. - It means "I know I should add this argument but I can't for BC"

**Comment by nicolas-grekas** [comment]
Actually I'm proposing to remove the argument from the interface in #31950
I fail to understand why it should be on the interface.

File: src/Symfony/Component/OptionsResolver/Options.php [hunk 1]
```diff
@@ -11,14 +11,34 @@
...
+     * @param bool   $triggerDeprecation Whether to trigger the deprecation or not
+     *
+     * @return mixed The option value
+     *
+     * @throws AccessException           If accessing this method outside of
+     *                                   {@link resolve()}
+     * @throws NoSuchOptionException     If the option is not set
+     * @throws InvalidOptionsException   If the option doesn't fulfill the
+     *                                   specified validation rules
+     * @throws OptionDefinitionException If there is a cyclic dependency between
```
Comment (nicolas-grekas) [inline src/Symfony/Component/OptionsResolver/Options.php]: all this is not a minor change: was it implicitly the definition of the contracts of the interface before?
also, does it make sense to have all this at the abstraction level? or are they implementation details of OptionsResolver, the implementation?

**Pull request closed without merging**